File: September/Day3/3.cpp
class Solution {
public:
    void MakeZeros(vector<vector<int> >& matrix) {
        // Code here
        int n = matrix.size();
        int m = matrix[0].size();
        vector<vector<int> > matcop;
        // for(int i=0; i<n;i++){
        //     for(int j=0; j<m;j++){
        //         matcop[i][j]= matrix[i][j];
                
        //     }
        // }
        matcop = matrix;
        
        for(int i=0; i<n;i++){
            
            for(int j=0; j<m;j++){
                if (matcop[i][j] == 0){
                    if(i-1 >=0){ 
                        matrix[i][j]  +=  matcop[i-1][j];
                         matrix[i-1][j] =0;}
                         
                    if(i+1 < n){ 
                        matrix[i][j]  +=  matcop[i+1][j];
                        matrix[i+1][j] =0;}
                         
                    if(j-1 >=0){ 
                        matrix[i][j]  +=  matcop[i][j-1];
                         matrix[i][j-1] =0;}
                         
                    if(j+1 < m){ 
                        matrix[i][j]  +=  matcop[i][j+1];
                         matrix[i][j+1] =0;}

                }
                
                
            }
        }
        
        
    }
};

File: September/Day3/3.py
import copy
class Solution:
	def MakeZeros(self, matrix):
		# Code here
		a = len(matrix)
		b = len(matrix[0])
		temp = copy.deepcopy(matrix) 
# 		Copying the matrix exact
        for i in range(a):
            for j in range(b):
                if (temp[i][j]==0):
                    if (i-1)>=0:
                        matrix[i][j] += temp[i-1][j]
                        matrix[i-1][j] =0
                        
                    if (i+1)<a:
                        matrix[i][j] += temp[i+1][j]
                        matrix[i+1][j] =0
                        
                    if (j-1)>=0:
                        matrix[i][j] += temp[i][j-1]
                        matrix[i][j-1] =0
                        
                    if (j+1)<b:
                        matrix[i][j] += temp[i][j+1]
                        matrix[i][j+1] =0
                        
        return matrix
